eXo/Sources/Controllers/Setting/SettingViewController.swift

```diff
@@ -0,0 +1,117 @@
+        super.viewDidLoad()
+        self.navigationItem.rightBarButtonItem = self.editButtonItem()
+        // Do any additional setup after loading the view.
+    }
+    
+    override func viewDidAppear(animated: Bool) {
+         self.tableView.reloadData()
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+
+    /*
+    // MARK: - Navigation
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    */
+    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
+        let indexPath = self.tableView.indexPathForSelectedRow
+        let server:Server = ServerManager.sharedInstance.serverList![(indexPath?.row)!]
+         as! Server
+        let serverEditVC:ServerEditViewController =  segue.destinationViewController as! ServerEditViewController
+        serverEditVC.server = server
+    }
+
+
+    
+    /*
+    // MARK: - Table View Datasource & Delegate
+    */
+    
+    override func numberOfSectionsInTableView(tableView: UITableView) -> Int{
+        return 1
+    }
+    
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if ( ServerManager.sharedInstance.serverList != nil) {
+            return ServerManager.sharedInstance.serverList!.count
+        }
+        return 0;
+    }
+    
+    override func tableView(tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return kHeaderHeight
+    }
+    
+    override func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        return kCellHeight
+    }
+    
+    
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        
+        let cell = tableView.dequeueReusableCellWithIdentifier("ServerCell", forIndexPath: indexPath)
+        cell.textLabel?.text = (ServerManager.sharedInstance.serverList?[indexPath.row] as! Server).serverURL.stringURLWithoutProtocol()
+        return cell
+    }
+    
+    override func tableView(tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return NSLocalizedString("Setting.Title.Server",comment:"")
+    }
+    
+    override func tableView(tableView: UITableView, canEditRowAtIndexPath indexPath: NSIndexPath) -> Bool {
+        return true;
+    }
+    
+    override func tableView(tableView: UITableView, commitEditingStyle editingStyle: UITableViewCellEditingStyle, forRowAtIndexPath indexPath: NSIndexPath) {
+        if editingStyle == UITableViewCellEditingStyle.Delete {
+
+            let alertController = UIAlertController(title: NSLocalizedString("Setting.Title.DeleteServer", comment: ""), message: NSLocalizedString("Setting.Message.DeleteServer", comment: ""), preferredStyle: UIAlertControllerStyle.Alert)
```

Comment:
This code should be in a separate method.
Besides, after the server is deleted, the Select Server screen is not updated, i.e. the deleted server may still appear in the most recent section. It should be updated to display the "new" most recent server.